edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/metrics/Avaliability.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class Availability
+    {
+        readonly ISystemTime time;
+
+        public string Name { get; private set; }
+        public TimeSpan RunningTime { get; private set; } = TimeSpan.Zero;
```

Comment:
Nit: why do we need these private setters?

Reply:
yes:
```
            this.ExpectedTime += duration;
            if (isUp)
            {
                this.RunningTime += duration;
            }
```